proto/auth.proto

```diff
@@ -69,8 +69,37 @@ message _GenerateApiTokenRequest {
         Expires expires = 2;
     }
 
-    // session token recieved from `momento login` command
-    string session_token = 3;
+    // Aliases for categories of functionality.
+    enum Role {
+      PermitNone = 0;
+      // Restricts access to apis that read and write data: No higher level resource description or modification.
+      ReadWrite = 1;
+    }
+
+    string auth_token = 3;
+
+    // Contains the permissions that the token will have for each of our services. If it doesn't exist
+    // on the request, then the token is assumed to have "SuperUser" permissions
+    message Permissions {
```

Comment:
We should probably just name this `Restrictions` since the inner content are all specific types of restrictions not actual permissions (allowed things).